pkg/agent/datamodel/sig_config.go

```diff
@@ -12,6 +16,136 @@ const (
 	AzurePublicCloudSigSubscription string = "109a5e88-712a-48ae-9078-9ca8b3c81345" // AKS VHD
 )
 
+//nolint:gochecknoglobals
+var (
+	CachedFromComponentContainerImages = make(map[string]ContainerImage)
+	CachedFromComponentDownloadedFiles = make(map[string]DownloadFiles)
+	CachedFromManifest                 = &Manifest{}
+)
+
+//nolint:gochecknoinits
+func init() {
+	manifest := CacheManifest()
+	processManifest(manifest)
+
+	components := CacheComponents()
+	processComponents(components)
+}
+
+func CacheManifest() Manifest {
+	_, filename, _, _ := runtime.Caller(0)
+	manifestFilePath := "../../../parts/linux/cloud-init/artifacts/manifest.json"
+	return getCachedVersionsFromManifestJSON(path.Join(path.Dir(filename), manifestFilePath))
+}
+
+func CacheComponents() Components {
+	_, filename, _, _ := runtime.Caller(0)
+	componentsFilePath := "../../../vhdbuilder/packer/components.json"
+	return getCachedVersionsFromComponentsJSON(path.Join(path.Dir(filename), componentsFilePath))
+}
+
+func processManifest(manifest Manifest) {
+	CachedFromManifest.Kubernetes = manifest.Kubernetes
+	CachedFromManifest.Runc = manifest.Runc
+	CachedFromManifest.Containerd = manifest.Containerd
+	CachedFromManifest.NvidiaContainerRuntime = manifest.NvidiaContainerRuntime
+	CachedFromManifest.NvidiaDrivers = manifest.NvidiaDrivers
+}
+
+func processComponents(components Components) {
+	for _, image := range components.ContainerImages {
+		componentName, err := processContainerImage(image.DownloadURL)
+		if err != nil {
+			panic(err)
+		}
+		CachedFromComponentContainerImages[componentName] = image
+	}
+	for _, file := range components.DownloadFiles {
+		componetName, err := processDownloadFile(file.DownloadURL)
+		if err != nil {
+			panic(err)
+		}
+		CachedFromComponentDownloadedFiles[componetName] = file
+	}
+}
+
+func getCachedVersionsFromManifestJSON(manifestFilePath string) Manifest {
+	data, err := os.ReadFile(manifestFilePath)
+	if err != nil {
+		panic(err)
+	}
+	data = bytes.ReplaceAll(data, []byte("#EOF"), []byte(""))
+	var manifest Manifest
+	if err = json.Unmarshal(data, &manifest); err != nil {
+		panic(err)
+	}
+	return manifest
+}
+
+func getCachedVersionsFromComponentsJSON(componentsFilePath string) Components {
+	data, err := os.ReadFile(componentsFilePath)
+	if err != nil {
+		panic(err)
+	}
+	var components Components
+	if err = json.Unmarshal(data, &components); err != nil {
+		panic(err)
+	}
+	return components
+}
+
+type CachedOnVHD struct {
+	CachedFromManifest                 *Manifest                 `json:"cached_from_manifest"`
+	CachedFromComponentContainerImages map[string]ContainerImage `json:"cached_from_component_container_images"`
+	CachedFromComponentDownloadedFiles map[string]DownloadFiles  `json:"cached_from_component_downloaded_files"`
+}
+
+type Dependency struct {
+	FileName         string            `json:"fileName"`
```

Comment:
feel free to omit FileName/DownloadLocation/DownloadURL if you're not using them

Reply:
Do you think there's any value for passing these along - maybe not for this rp ask, but for anything down the line? Otherwise I can remove FileName, DownloadLocation, DownloadURL

Reply:
I don't really think there's much value, I'd probably prefer omitting them

we can always add them back later if we think we need them